src/SeedCppCore/REST/RouteAccess/RouteAccessValidatorsFactory.cpp

```diff
@@ -23,9 +25,9 @@ namespace seed_cpp { namespace rest {
 	std::unique_ptr<IRouteAccessValidator> RouteAccessValidatorsFactory::buildTokenExpirationRouteAccessValidator() const
 	{
 		long expirationSeconds = 600; // 10 minutes
-		auto& epochTimeService = m_context.getServicesMgr()->getSystemServicesMgr().getEpochTimeService();
+		systelab::time::TimeAdapter timeAdapter;
```

Comment:
The TimeAdapter shouldn't be declared as a local variable because it is injected to the service being built, but it will be destroyed once exiting the factory method. Thus, this will cause a crash when accessing to an object that has been destroyed from the service.

Thus, I would suggest defining the TimeAdapter as part of the context, as we do with other adapters (i.e db adapter). Moreover, it should be injected on constructor of the Context object and as an interface (ITimeAdapter), so we allow using a mock version of it (we are not assuming to have always the real implementation).